packages/analytics/shippers/elastic_v3/browser/src/browser_shipper.ts

```diff
@@ -125,6 +125,10 @@ export class ElasticV3BrowserShipper implements IShipper {
   }
 
   private async makeRequest(events: Event[]): Promise<string> {
+    if (events.length === 0) {
+      return '0'; // Nothing to send.
+    }
+
```

Comment:
Given both methods are private, so we don't have risk to have any external consumer calling the method by mistake, my gut feeling would have been to perform this check in the `sendEvents` method instead.

```ts
  private async sendEvents(events: Event[]) {
    if(!events.length) {
        return; 
    }
    try {
      const code = await this.makeRequest(events);
      this.reportTelemetryCounters(events, { code });
    } catch (error) {
      this.reportTelemetryCounters(events, { code: error.code, error });
    }
  }
```

Main upside, in addition to a proper isolation of concerns, would be to avoid calling `reportTelemetryCounters` unnecessarily when called with an empty list of event (unless that's a behavior we actually want)

Same goes for the server-side shipper.

Reply:
FWIW, I agree with the gut feeling. 

Being private, they shouldn't be called with empty arrays. I think I'll rework both solutions and avoid the `if(events.length)` in the internal methods altogether.

For once, the browser_shipper already has that in https://github.com/elastic/kibana/blob/19e84d96aca45a52958f4396e420105620b63c66/packages/analytics/shippers/elastic_v3/browser/src/browser_shipper.ts#L110-L113
So it's not really necessary downstream.

I'll rework `server_shipper` as well.